stl/inc/format

```diff
@@ -1248,15 +1243,425 @@ _OutputIt _Write(_OutputIt _Out, basic_string_view<_CharT> _Value) {
+            _Fill_left  = (_Specs._Width - _Width) / 2;
+            _Fill_right = _Specs._Width - _Width - _Fill_left;
+            break;
+        default:
+            _STL_ASSERT(false, "Invalid alignment");
+        }
+    }
+
+    // TRANSITION, add support for unicode/wide formats
+    _Out = _STD fill_n(_Out, _Fill_left, _Specs._Fill[0]);
+    _Out = _Fn(_Out);
+    return _STD fill_n(_Out, _Fill_right, _Specs._Fill[0]);
+}
+
+template <integral _Integral>
+string_view _Get_integral_prefix(char _Type, _Integral _Value) {
+    switch (_Type) {
+    case 'b':
+        return "0b"sv;
+    case 'B':
+        return "0B"sv;
+    case 'x':
+        return "0x"sv;
+    case 'X':
+        return "0X"sv;
+    case 'o':
+        if (_Value != _Integral{0})
+            return "0"sv;
+    }
+    return {};
+}
+
+template <class _OutputIt>
+_OutputIt _Write_sign(_OutputIt _Out, _Sign _Sgn, bool _Is_negative) {
+    if (_Is_negative) {
+        _Out = '-';
+    } else {
+        switch (_Sgn) {
+        case _Sign::_Plus:
+            _Out = '+';
+            break;
+
+        case _Sign::_Space:
+            _Out = ' ';
+            break;
+        }
+    }
+    return ++_Out;
+}
+
+inline void _Fmt_buffer_to_uppercase(char* _Begin, const char* _End) {
+    for (auto _It = _Begin; _It < _End; ++_It) {
+        *_It = static_cast<char>(_CSTD toupper(*_It));
+    }
+}
+
+template <class _CharT, class _Ty>
+bool _In_bounds(_Ty _Value) {
+    if constexpr (is_unsigned_v<_CharT> && is_unsigned_v<_Ty>) {
+        return _Value <= (numeric_limits<_CharT>::max)();
+    } else if constexpr (is_unsigned_v<_CharT>) {
+        return _Value >= 0 && static_cast<make_unsigned_t<_Ty>>(_Value) <= (numeric_limits<_CharT>::max)();
+    } else if constexpr (is_unsigned_v<_Ty>) {
+        return _Value <= static_cast<make_unsigned_t<_CharT>>((numeric_limits<_CharT>::max)());
+    } else {
+        return (numeric_limits<_CharT>::min)() <= _Value && _Value <= (numeric_limits<_CharT>::max)();
+    }
+}
+
+template <class _CharT, class _OutputIt>
+_OutputIt _Write(_OutputIt _Out, monostate, _Basic_format_specs<_CharT>&) {
     _STL_INTERNAL_CHECK(false);
-    (void) _Val;
-    (void) _Specs;
     return _Out;
 }
 
+#pragma warning(push)
+#pragma warning(disable : 4365) // 'argument': conversion from 'char' to 'const wchar_t', signed/unsigned mismatch
+template <class _CharT, class _OutputIt, integral _Integral>
+_OutputIt _Write_integral(_OutputIt _Out, _Integral _Value, _Basic_format_specs<_CharT>& _Specs) {
+    if (_Specs._Type && _Specs._Type == 'c') {
+        if (!_In_bounds<_CharT>(_Value))
+            throw format_error("integral cannot be stored in charT");
+        _Specs._Alt = false;
+        return _Write(_Out, static_cast<_CharT>(_Value), _Specs);
+    }
+
+    if (_Specs._Precision != -1)
+        throw format_error("integral cannot have a precision");
+
+    if (_Specs._Sgn == _Sign::_None)
+        _Specs._Sgn = _Sign::_Minus;

```

Comment:
The default sign is `-`, so I am trying to model that. It also makes the logic lower for adding a width to positive numbers a bit simpler (instead of comparing against `_Minus` and `_None`, we only compare to `_Minus`).